packages/balancer/skills/pool_manager_abci/rounds.py

```diff
@@ -186,3 +181,15 @@ class PoolManagerAbciApp(AbciApp[Event]):
         Event.ROUND_TIMEOUT: 30.0,
     }
     cross_period_persisted_keys: List[str] = []
+    db_pre_conditions: Dict[AppState, List[str]] = {
+        DecisionMakingRound: [
+            get_name(SynchronizedData.most_voted_estimates),
+        ],
+    }
+    db_post_conditions: Dict[AppState, List[str]] = {
+        FinishedTxPreparationRound: [
+            get_name(SynchronizedData.most_voted_tx_hash),
+            get_name(SynchronizedData.safe_contract_address),
```

Comment:
The `safe_contract_address` is not "produced" in this skill, so shouldn't be a post-condition.

Reply:
There's an error if I don't add it here. It seems the `FinishedRegistrationRound` branch is not being traversed. 
```
Error: Pre conditions '{'safe_contract_address'}' of app '<class 'packages.valory.skills.transaction_settlement_abci.rounds.TransactionSubmissionAbciApp'>' not a post condition of app '<class 'packages.balancer.skills.pool_manager_abci.rounds.PoolManagerAbciApp'>' or any preceding app in path [(<class 'packages.valory.skills.registration_abci.rounds.RegistrationStartupRound'>, <class 'packages.valory.skills.registration_abci.rounds.AgentRegistrationAbciApp'>, <class 'packages.valory.skills.registration_abci.rounds.FinishedRegistrationFFWRound'>), (<class 'packages.balancer.skills.fear_and_greed_oracle_abci.rounds.ObservationRound'>, <class 'packages.balancer.skills.fear_and_greed_oracle_abci.rounds.FearAndGreedOracleAbciApp'>, <class 'packages.balancer.skills.fear_and_greed_oracle_abci.rounds.FinishedDataCollectionRound'>), (<class 'packages.balancer.skills.pool_manager_abci.rounds.DecisionMakingRound'>, <class 'packages.balancer.skills.pool_manager_abci.rounds.PoolManagerAbciApp'>, <class 'packages.balancer.skills.pool_manager_abci.rounds.FinishedTxPreparationRound'>), (<class 'packages.valory.skills.transaction_settlement_abci.rounds.RandomnessTransactionSubmissionRound'>, <class 'packages.valory.skills.transaction_settlement_abci.rounds.TransactionSubmissionAbciApp'>, <class 'packages.valory.skills.transaction_settlement_abci.rounds.FinishedTransactionSubmissionRound'>), (<class 'packages.valory.skills.reset_pause_abci.rounds.ResetAndPauseRound'>, <class 'packages.valory.skills.reset_pause_abci.rounds.ResetPauseAbciApp'>, None)].

```

Reply:
Then open an issue please

Reply:
https://github.com/valory-xyz/open-autonomy/issues/1681

Reply:
Looks to me like the same issue as here: https://github.com/valory-xyz/price-oracle/pull/24/files - although still need to confirm.

Reply:
Looks the same:

```
Error: Pre conditions '{'safe_contract_address'}' of app '<class 'packages.valory.skills.transaction_settlement_abci.rounds.TransactionSubmissionAbciApp'>' not a post condition of app '<class 'packages.balancer.skills.pool_manager_abci.rounds.PoolManagerAbciApp'>' or any preceding app in path

[
(<class 'packages.valory.skills.registration_abci.rounds.RegistrationStartupRound'>, <class 'packages.valory.skills.registration_abci.rounds.AgentRegistrationAbciApp'>, <class 'packages.valory.skills.registration_abci.rounds.FinishedRegistrationFFWRound'>),

(<class 'packages.balancer.skills.fear_and_greed_oracle_abci.rounds.ObservationRound'>, <class 'packages.balancer.skills.fear_and_greed_oracle_abci.rounds.FearAndGreedOracleAbciApp'>, <class 'packages.balancer.skills.fear_and_greed_oracle_abci.rounds.FinishedDataCollectionRound'>),

(<class 'packages.balancer.skills.pool_manager_abci.rounds.DecisionMakingRound'>, <class 'packages.balancer.skills.pool_manager_abci.rounds.PoolManagerAbciApp'>, <class 'packages.balancer.skills.pool_manager_abci.rounds.FinishedTxPreparationRound'>),

(<class 'packages.valory.skills.transaction_settlement_abci.rounds.RandomnessTransactionSubmissionRound'>, <class 'packages.valory.skills.transaction_settlement_abci.rounds.TransactionSubmissionAbciApp'>, <class 'packages.valory.skills.transaction_settlement_abci.rounds.FinishedTransactionSubmissionRound'>),

(<class 'packages.valory.skills.reset_pause_abci.rounds.ResetAndPauseRound'>, <class 'packages.valory.skills.reset_pause_abci.rounds.ResetPauseAbciApp'>, None)
].
```
this path is not possible without specifying the appropriate post condition of the registration skill.